include/net/zzhc.rst

```diff
@@ -0,0 +1,74 @@
+.. _lib_zzhc:
+
+China Telecom ZZHC library
+#################################################
+
+The China Telecom ZZHC library implements the self-registration (Zi-ZHu-Ce, ZZHC) functionality that registers the module information on the server when the network is available.
+
+Overview
+********
+
+The telecommunication modules that are part of the China Telecom network are required to provide certain information stored on the module to the `zzhc.vnet.cn` server.
+This is required for conformity with Chapter 6 of the China Telecom IoT Module Requirements Whitepaper.
+
+The ZZHC library is started as a background thread on boot and it automatically detects the availability of the operator network.
+Once the Internet connectivity is ready and a new SIM card is detected on boot, the library proactively uploads the following information to the ZZHC server:
+- IMEI
+- model number (that is, nRF9160)
+- modem revision
+- nRF9160 SIM ICCID
+- nRF9160 SIM IMSI
+On any failure, which include unknown/negative response from server or any error during program execution, the ZZHC library retries the next hour for at most 5 times.
+
+.. _lib_zzhc_dependencies:
+
+Dependencies
+************
+
+The ZZHC library only when CONFIG_TRUSTED_EXECUTION_NONSECURE is enabled (i.e. Trusted Execution: Non-Secure firmware image).
+
+The ZZHC library requires the following to operate, therefore they are automatically selected when ZZHC library is enabled.
+   * CONFIG_BSD_LIBRARY=y
+   * CONFIG_SETTINGS=y
+   * CONFIG_AT_CMD=y
+   * CONFIG_AT_CMD_PARSER=y
+   * CONFIG_AT_NOTIF=y
+   * CONFIG_BASE64=y
+   * CONFIG_JSON_LIBRARY=y
+
+.. _lib_zzhc_configuration:
+
+Configurations
+**************
+
+The ZZHC library is by default not used in any sample.
+To enable the library so that the end product can be approved by China Telecom, enable Kconfig option CONFIG_ZZHC.
+In prj.conf, set CONFIG_ZZHC=y.
+
+The thread behavior can also be configured using Kconfig options:
+   * To configure the stack size for the thread, change CONFIG_ZZHC_STACK_SIZE.
+   * To configure the thread priority, change CONFIG_ZZHC_THREAD_PRIO.
+
+The minimal heap size required is 4096 bytes.
+In prj.conf, set CONFIG_HEAP_MEM_POOL_SIZE=4096.
+
+Among the depended libraries, CONFIG_BSD_LIBRARY and CONFIG_SETTINGS require further configurations in prj.conf.
+
+Required by CONFIG_BSD_LIBRARY:
+   * CONFIG_NETWORKING=y
+   * CONFIG_NET_NATIVE=n
+   * CONFIG_NET_SOCKETS=y
+   * CONFIG_NET_SOCKETS_OFFLOAD=y
+
+Required by CONFIG_SETTINGS:
+   * CONFIG_FLASH=y
+   * CONFIG_FLASH_PAGE_LAYOUT=y
+   * CONFIG_FLASH_MAP=y
+   * CONFIG_MPU_ALLOW_FLASH_WRITE=y
+   * CONFIG_NVS=y
+   * CONFIG_SETTINGS_NVS_SECTOR_COUNT=6
+
+To allow for automatic registration to Lte-M or NB-IoT network on boot, the following are required in prj.conf:
+   * CONFIG_LTE_LINK_CONTROL=y
+   * CONFIG_LTE_AUTO_INIT_AND_CONNECT=y
+
```

Comment:
Just checking: Is there going to be any API documentation appended to this document? All other libraries have the API documentation section included in their respective rst files.

Reply:
No. This zzhc library does not have public API, so there is no documentation.

Instead, it's just a background thread which sends critical information to cloud quietly.